README.md

```diff
@@ -1 +1,151 @@
-# angular-creating-custom-filters-readme
\ No newline at end of file
+# Creating custom Filters
+
+## Overview
+
+Just like services and directives, we can also create custom filters to use in our application.
+
+## Objectives
+
+- Describe custom filters
+- Create a custom filter
+- Use a custom filter with a single value
+- Use a custom filter with a dataset
+
+## Custom Filters
+
+Custom filters come in two types - single value filters and dataset filters. We can apply a custom filter to filter an array in an `ng-repeat`, or we can apply a filter to an expression to format the expression differently. Both of these are declared via the `.filter` method.
+
+```js
+function MyCustomFilter() {
+	// wooo!
+}
+
+angular
+	.module('app')
+	.filter('myCustomFilter', MyCustomFilter);
+```
+
+Looks familiar, yes?
+
+Our custom filters should return a function that will then process the data passed to it. Let's create a filter to make our expression uppercase.
+
+## Single value filters
+
+When we return a function in our filter, as such:
+
+```js
+function MyCustomFilter() {
+	return function (str) {
+
+	};
+}
+
+angular
+	.module('app')
+	.filter('myCustomFilter', MyCustomFilter);
+```
+
+We get the expression's value passed through to the function. Anything we then returned from our returned function will be put in place of the expression. We could return the string `'hello'` but that wouldn't be any use to anyone.
+
+Let's return the `str` variable, in uppercase.
+
+```js
+function makeUppercase() {
+	return function (str) {
+		return str.toUpperCase();
+	};
+}
+
+angular
+	.module('app')
+	.filter('makeUppercase', makeUppercase);
+```
+
+Awesome! We can now use this like any other filter in Angular.
+
+```html
+<div>
+	{{ 'My String' | makeUppercase }}
+
+	{{ ctrl.variable | makeUppercase }}
+</div>
+```
+
+## Arguments
+
+We can also pass arguments through to our custom filters. Much like what we've done with the `date` filter before, we can pass through as many arguments we like to the filter, and they're then the 2nd, 3rd, etc, argument passed through to our function.
+
+We can add a toggle to our `makeUppercase` filter on whether or not to *actually* make the text uppercase.
+
+```html
+<div>
+	{{ 'My String' | makeUppercase:true }}
+
+	{{ ctrl.variable | makeUppercase:false }}
```

Comment:
this is sort of a weird example - why would you apply the filter at all just to set it to false? How might you use this in a real application? My instinct is that you might have some type of conditional here